File: generate-rss.js
const fs = require('fs');
const axios = require('axios');

async function fetchWatchlist() {
    const response = await axios.get('https://api.trakt.tv/users/me/watchlist/shows', {
        headers: {
            'Authorization': `Bearer ${process.env.TRAKT_ACCESS_TOKEN}`,
            'Content-Type': 'application/json',
            'trakt-api-version': '2',
            'trakt-api-key': process.env.TRAKT_CLIENT_ID
        }
    });
    return response.data.map(item => ({
        imdbId: item.show.ids.imdb.replace('tt', ''),
        title: item.show.title
    }));
}

async function fetchTorrentsForWatchlist() {
    const watchlist = await fetchWatchlist();
    const torrents = [];
    for (const item of watchlist) {
        const res = await axios.get(`https://eztvx.to/api/get-torrents?imdb_id=${item.imdbId}&limit=10&page=1`);
        if (res.data.torrents) {
            torrents.push(...res.data.torrents.map(torrent => ({
                ...torrent,
                showTitle: item.title
            })));
        }
    }
    return torrents;
}

function parseTorrentTitle(title) {
    const regex = /^(.*?)\s*S(\d{1,2})E(\d{1,2})(?:\s*(.*?))?\s*(2160p|1080p|720p|480p)?\s*(x264|x265)?\s*(?:-(\w+))?$/i;
    const match = title.match(regex);
    if (match) {
        return {
            showName: match[1].trim(),
            season: parseInt(match[2], 10),
            episode: parseInt(match[3], 10),
            episodeTitle: match[4] ? match[4].trim() : '',
            quality: match[5] ? match[5].trim() : '',
            encoding: match[6] ? match[6].trim() : '',
            team: match[7] ? match[7].trim() : ''
        };
    }
    return null;
}

function formatFileSize(bytes) {
    if (bytes >= 1073741824) return (bytes / 1073741824).toFixed(2) + ' GB';
    else if (bytes >= 1048576) return (bytes / 1048576).toFixed(2) + ' MB';
    else if (bytes >= 1024) return (bytes / 1024).toFixed(2) + ' KB';
    else return bytes + ' Bytes';
}

async function generateRSS() {
    const torrents = await fetchTorrentsForWatchlist();
    let xml = `<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0">
    <channel>
        <title>Watchlist Torrents</title>
        <link>https://yourusername.github.io/your-repo/</link>
        <description>تورنتات من قائمة المراقبة في Trakt.tv</description>
        <language>ar</language>`;

    torrents.forEach(torrent => {
        const parsed = parseTorrentTitle(torrent.title);
        if (parsed) {
            xml += `
        <item>
            <title>${parsed.showName} S${parsed.season.toString().padStart(2, '0')}E${parsed.episode.toString().padStart(2, '0')}${parsed.episodeTitle ? ' - ' + parsed.episodeTitle : ''}</title>
            <link>${torrent.magnet_url}</link>
            <description>جودة: ${parsed.quality || 'غير محدد'} | حجم: ${formatFileSize(torrent.size_bytes)}</description>
            <pubDate>${new Date(torrent.date_released_unix * 1000).toUTCString()}</pubDate>
            <guid>${torrent.magnet_url}</guid>
        </item>`;
        }
    });

    xml += `
    </channel>
</rss>`;
    fs.writeFileSync('rss.xml', xml);
}

generateRSS().catch(console.error);

File: generate_rss.py
#!/usr/bin/env python3
import requests
import xml.etree.ElementTree as ET
from xml.dom import minidom
import argparse

def format_file_size(bytes):
    """تحويل الحجم من البايت إلى الوحدة المناسبة"""
    if bytes >= 1073741824:
        return f"{bytes / 1073741824:.2f} GB"
    elif bytes >= 1048576:
        return f"{bytes / 1048576:.2f} MB"
    elif bytes >= 1024:
        return f"{bytes / 1024:.2f} KB"
    else:
        return f"{bytes} Bytes"

def fetch_torrents(imdb_id, page=1, results_per_page=30):
    """استدعاء API لجلب التورنتات بناءً على معرف IMDb ورقم الصفحة"""
    url = f"https://eztvx.to/api/get-torrents?imdb_id={imdb_id}&limit={results_per_page}&page={page}"
    response = requests.get(url)
    response.raise_for_status()
    return response.json()

def generate_rss_feed(torrents, feed_title="Torrent RSS Feed"):
    """بناء خلاصة RSS من بيانات التورنتات"""
    rss = ET.Element("rss", version="2.0")
    channel = ET.SubElement(rss, "channel")
    ET.SubElement(channel, "title").text = feed_title
    ET.SubElement(channel, "description").text = "خلاصة RSS لنتائج تورنت البحث"
    ET.SubElement(channel, "link").text = "https://yourwebsite.com"  # عدل الرابط حسب الحاجة

    for torrent in torrents:
        item = ET.SubElement(channel, "item")
        # استخدام عنوان التورنت كنص للعنصر
        title_text = torrent.get("title", "No Title")
        ET.SubElement(item, "title").text = title_text
        # وصف يتضمن حجم التورنت
        size = torrent.get("size_bytes", 0)
        ET.SubElement(item, "description").text = f"الحجم: {format_file_size(size)}"
        # الرابط الخاص بالتحميل (magnet)
        ET.SubElement(item, "link").text = torrent.get("magnet_url", "")
        # استخدام معرف التورنت أو العنوان كـ GUID
        guid = torrent.get("id", title_text)
        ET.SubElement(item, "guid").text = str(guid)
    
    # تنسيق XML بشكل منسق (pretty-print)
    xml_str = ET.tostring(rss, encoding='utf-8')
    parsed = minidom.parseString(xml_str)
    pretty_xml_str = parsed.toprettyxml(indent="  ")
    return pretty_xml_str

def main():
    parser = argparse.ArgumentParser(description="توليد خلاصة RSS من نتائج تورنت البحث")
    parser.add_argument("--imdb", required=True, help="معرف IMDb بدون بادئة 'tt'")
    parser.add_argument("--page", type=int, default=1, help="رقم الصفحة لنتائج البحث")
    parser.add_argument("--output", default="feed.xml", help="اسم ملف الخلاصة الناتج")
    args = parser.parse_args()

    results_per_page = 30
    data = fetch_torrents(args.imdb, page=args.page, results_per_page=results_per_page)
    torrents = data.get("torrents", [])
    
    if not torrents:
        print("لم يتم العثور على نتائج للتورنت.")
        return

    rss_feed = generate_rss_feed(torrents)
    
    with open(args.output, "w", encoding="utf-8") as f:
        f.write(rss_feed)
    print(f"تم إنشاء خلاصة RSS بـ {len(torrents)} عنصر وحفظها في {args.output}")

if __name__ == "__main__":
    main()